packages/docusaurus-theme-classic/src/theme/Navbar/Search/index.tsx

```diff
@@ -7,13 +7,18 @@
 
 import React from 'react';
 import clsx from 'clsx';
+import useDocusaurusContext from '@docusaurus/useDocusaurusContext';
 import type {Props} from '@theme/Navbar/Search';
 
 import styles from './styles.module.css';
 
 export default function NavbarSearch({
   children,
   className,
-}: Props): JSX.Element {
+}: Props): JSX.Element | null {
+  const {siteConfig} = useDocusaurusContext();
+  if (!siteConfig.themeConfig?.algolia) {
```

Comment:
this will not work for non-official search plugins

Reply:
em...Let's see if there's another way.